src/kernels/jupyter/session/jupyterRequestCreator.web.ts

```diff
@@ -80,8 +81,38 @@ export class JupyterRequestCreator implements IJupyterRequestCreator {
                     traceError('KernelId not extracted from Kernel WebSocket URL');
                 }
 
+                // TODO: Implement ping. Well actually see if ping is necessary
                 // Ping the websocket connection every 30 seconds to make sure it stays alive
                 //timer = this.timer = setInterval(() => this.ping(), 30_000);
+
+                // On open, replace the onmessage handler with our own.
+                this.addEventListener('open', this.boundOpenHandler);
+            }
+
+            private openHandler() {
```

Comment:
This code here is how on web, messages are 'emitted' still. I fudge the onmessage handler and turn it into an emit like the node websocket does.

This allows the emit hooks for ipywidgets to fire.